cli/cmd/sync.go

```diff
@@ -113,5 +131,72 @@ func copyIntegrationsComposeFiles(beats git.Project, target string) {
 				"_meta": metaDir,
 			}).Warn("Meta dir was not copied")
 		}
+
+		err = sanitizeComposeFile(targetFile)
+		if err != nil {
+			log.Fatalf("error: %v", err)
+		}
 	}
 }
+
+type service interface{}
+type compose struct {
+	Version  string             `yaml:"version"`
+	Services map[string]service `yaml:"services"`
+}
+
+// removes non-needed blocks in the target compose file, such as the build context
+func sanitizeComposeFile(composeFilePath string) error {
+	bytes, err := io.ReadFile(composeFilePath)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"docker-compose": composeFilePath,
+		}).Error("Could not read docker compose file")
+		return err
+	}
+
+	c := compose{}
+	err = yaml.Unmarshal(bytes, &c)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"docker-compose": composeFilePath,
+		}).Error("Could not unmarshal docker compose file")
+		return err
+	}
+
+	// we'll copy all fields but the excluded ones in this struct
+	output := make(map[string]service)
+
+	for k, srv := range c.Services {
+		switch i := srv.(type) {
+		case map[interface{}]interface{}:
+			log.WithFields(log.Fields{
+				"name":         k,
+				"compose-file": composeFilePath,
+			}).Debug("sanitize service in docker-compose file")
+
+			for key, value := range i {
+				strKey := fmt.Sprintf("%v", key)
+
+				// remove the build context element
+				if contains(excludedBlocks, strKey) {
+					continue
+				}
+
+				output[strKey] = value
+			}
+		default:
+			// skip
+		}
+
+		c.Services[k] = output
+	}
+
+	d, err := yaml.Marshal(&c)
+	if err != nil {
+		return err
+	}
+	io.WriteFile(d, composeFilePath)
```

Comment:
Could this silently fail if the user doesn't have permissions to write to the file?